main.go

```diff
@@ -18,18 +39,69 @@ func getPing(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func getBooks(w http.ResponseWriter, r *http.Request) {
+func Books(w http.ResponseWriter, r *http.Request) {
+	metodo := r.Method
+	fmt.Printf("metodo: %v\n", metodo)
+	switch {
+	case metodo == "GET":
+		getBooks(w, r)
+	case metodo == "POST":
+		postBooks(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
 
-	bookslist := []struct { //Struct created to handle data of the list of books
-		Name      string  `json:"name"`
-		Price     float32 `json:"price"`
-		Inventory int     `json:"inventory"`
-	}{
-		{"Book 1", 30.20, 2},
-		{"Book 2", 20.30, 1},
-		{"Book 3", 32.20, 5},
+func postBooks(w http.ResponseWriter, r *http.Request) {
+	//TO DO:
+	//Read the Json body
+	var newBook = Book{"sem nome", -1, -1, uuid.Nil}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&newBook)
```

Comment:
Usually we do it this way:
```suggestions
        err := json.NewDecoder(r.Body).Decode(&newBook)

Reply:
Ok.